internal/beater/server.go

```diff
@@ -191,13 +192,19 @@ func newServer(args ServerParams, listener net.Listener) (server, error) {
 		return server{}, err
 	}
 
-	otlpBatchProcessor := args.BatchProcessor
+	var otlpBatchProcessor modelprocessor.Chained
+
+	esClient, err := args.NewElasticsearchClient(elasticsearch.DefaultConfig())
```

Comment:
I couldn't find how the esClient instance is initialized for RUM sourcemaps. I found about this `DefaultConfig` and it allowed me to compile so that's why I used it, although I'm not sure if this is the proper way to initialize it, or if there's another way for us to get an instance of esClient without having to initialize it in here. I'm open to any advice on this!

Reply:
It happens here: https://github.com/elastic/apm-server/blob/main/internal/beater/beater.go#L340-L353

We're passing a callback to the method which creates the fetcher.

Reply:
Thanks for pointing me in the right direction. I just tried to replicate the same setup for the esClient instance as it's done in `beater.go`, however, I noticed that there are a couple of intricate setup parameters and processes that require creating tracers and callbacks and launching goroutines that seem too many things that I can do wrongly, so I then noticed that this code in `server.go` that adds the OTLP processors is called from `beater.go` from the same place that you mentioned, so I decided to pass the esClient initialization function and config from `beater.go` instead and that way not having to set it all up again in `server.go`. I'm unsure if I'm breaking any rules by doing so, so please have another look and let me know if there's something I need to change with that approach.